File: src/com/douma/written_test/meituan/_20210404/_5_Candy.py
# 抖码算法，让算法学习变得简单有趣
# 作者：老汤

if __name__ == '__main__':
    line = str(input()).split(" ")
    a, b = int(line[0]), int(line[1])

    line = str(input()).split(" ")
    candy_A = [0] * a
    for i in range(a):
        candy_A[i] = int(line[i])

    line = str(input()).split(" ")
    candy_B = [0] * b
    for i in range(b):
        candy_B[i] = int(line[i])

    # 1. 计算 A 的最大的前缀和
    max_A, prefix_A = 0, [0] * (a + 1)
    for i in range(1, a + 1):
        prefix_A[i] = prefix_A[i - 1] + candy_A[i - 1]
        max_A = max(max_A, prefix_A[i])

    # 2. 计算 B 的最大的前缀和
    max_B, prefix_B = 0, [0] * (b + 1)
    for i in range(1, b + 1):
        prefix_B[i] = prefix_B[i - 1] + candy_B[i - 1]
        max_B = max(max_B, prefix_B[i])

    print(max_A + max_B)

File: src/com/douma/笔试代码/meituan/_20210404/_5_Candy.java
package com.douma.笔试代码.meituan._20210404;

import java.util.Scanner;

/**
 * 抖码算法，让算法学习变的简单有趣
 *
 * @作者 : 老汤
 */
public class _5_Candy {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int a = scanner.nextInt();
        int b = scanner.nextInt();

        int[] candyA = new int[a];
        for (int i = 0; i < a; i++) {
            candyA[i] = scanner.nextInt();
        }

        int[] candyB = new int[b];
        for (int i = 0; i < b; i++) {
            candyB[i] = scanner.nextInt();
        }

        // 1. 计算 A 的最大的前缀和
        int maxA = 0;
        int[] prefixA = new int[a + 1];
        prefixA[0] = 0;
        for (int i = 1; i <= a; i++) {
            prefixA[i] = prefixA[i - 1] + candyA[i - 1];
            maxA = Math.max(maxA, prefixA[i]);
        }

        // 2. 计算 B 的最大的前缀和
        int maxB = 0;
        int[] prefixB = new int[b + 1];
        prefixB[0] = 0;
        for (int i = 1; i <= b; i++) {
            prefixB[i] = prefixB[i - 1] + candyB[i - 1];
            maxB = Math.max(maxB, prefixB[i]);
        }

        System.out.println(maxA + maxB);
    }
}
